### PR TITLE
[fix]Add  logging before _builtin_unreachable()

### PR DESCRIPTION
**Pull request by Tech-Circle-48**
## Further comments

Add FATAL logging before _builtin_unreachable()

### CODE REVIEW

File: be/src/pipeline/exec/hashjoin_build_sink.cpp [hunk 1]
```diff
@@ -173,6 +173,7 @@ Status HashJoinBuildSinkLocalState::process_build_block(RuntimeState* state,
                     [&](std::monostate& arg, auto has_null_value,
                         auto short_circuit_for_null_in_build_side) -> Status {
                         LOG(FATAL) << "FATAL: uninited hash table";
+                        LOG(FATAL) << "__builtin_unreachable";
```
Comment (mrhhsg) [inline be/src/pipeline/exec/hashjoin_build_sink.cpp:176]: Dumplicated fatal log.

File: be/src/pipeline/exec/hashjoin_build_sink.cpp [hunk 2]
```diff
@@ -355,6 +356,7 @@ void HashJoinBuildSinkLocalState::_hash_table_init(RuntimeState* state) {
 
     std::visit(vectorized::Overload {[&](std::monostate& arg) {
                                          LOG(FATAL) << "FATAL: uninited hash table";
+                                         LOG(FATAL) << "__builtin_unreachable";
```
Comment (mrhhsg) [inline be/src/pipeline/exec/hashjoin_build_sink.cpp:359]: Duplicated

File: be/src/pipeline/exec/hashjoin_build_sink.cpp [hunk 3]
```diff
@@ -496,6 +498,7 @@ Status HashJoinBuildSinkOperatorX::sink(RuntimeState* state, vectorized::Block*
         }
         auto ret = std::visit(Overload {[&](std::monostate&) -> Status {
                                             LOG(FATAL) << "FATAL: uninited hash table";
+                                            LOG(FATAL) << "__builtin_unreachable";
```
Comment (mrhhsg) [inline be/src/pipeline/exec/hashjoin_build_sink.cpp:501]: Dumplicated

File: be/src/pipeline/exec/hashjoin_build_sink.cpp [hunk 4]
```diff
@@ -552,6 +555,7 @@ Status HashJoinBuildSinkOperatorX::sink(RuntimeState* state, vectorized::Block*
                     Overload {
                             [&](std::monostate&) -> Status {
                                 LOG(FATAL) << "FATAL: uninited hash table";
+                                LOG(FATAL) << "__builtin_unreachable";
```
Comment (mrhhsg) [inline be/src/pipeline/exec/hashjoin_build_sink.cpp:558]: Duplicated

File: be/src/vec/columns/column.h [hunk 5]
```diff
@@ -152,6 +152,7 @@ class IColumn : public COW<IColumn> {
     // todo(Amory) from column to get data type is not correct ,column is memory data,can not to assume memory data belong to which data type
     virtual TypeIndex get_data_type() const {
         LOG(FATAL) << "Cannot get_data_type() column " << get_name();
+        LOG(FATAL) << "__builtin_unreachable";
```
Comment (mrhhsg) [inline be/src/vec/columns/column.h:155]: Duplicated

**Review by mrhhsg — CHANGES_REQUESTED** [review]
There are some duplicated fatal logs.

File: be/src/pipeline/exec/hashjoin_build_sink.cpp (hunk 2 above)
Comment (Tech-Circle-48) [reply be/src/pipeline/exec/hashjoin_build_sink.cpp:358]: already modilfy

Comment (Tech-Circle-48) [reply be/src/pipeline/exec/hashjoin_build_sink.cpp:358]: already modifed

**Review by yiguolei — APPROVED** [review]
lgtm

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Review by wm1581066 — APPROVED** [review]
LGTM

**Pull request merged**